src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/Validators/UriValidationBehavior.shared.cs

```diff
@@ -23,8 +25,8 @@ public UriKind UriKind
 			set => SetValue(UriKindProperty, value);
 		}
 
-		protected override bool Validate(object value)
-			=> base.Validate(value)
+		protected override async Task<bool> ValidateAsync(object value, CancellationToken token)
+			=> await base.ValidateAsync(value, token)
```

Comment:
can we just return a `Task.FromResult` here

Reply:
I think, no, cuz we should await the result from the base class.

Reply:
But we await the `ValidateAsync` somewhere, no?

Reply:
Yes, but I don't think we can rewrite it without awaiting
```csharp
await base.ValidateAsync(value, token) && Uri.IsWellFormedUriString(value?.ToString(), UriKind);
```

we need both values (one from the base class and another from here)

Reply:
Yes, the code where we have this snippet that you paste here, needs to be async this ValidadeAsync doesn't need. Or is this snippet just to illustrate the idea and doesn't exist on our code.

Reply:
@pictos now every developer can write custom class inherited from ValidationBehavior and define custom logic of validation that can be async.

As for our validators, they all don't need to be async

Reply:
So far you don't show me why we need this to be async. The method that should be async is the method that calls this `ValidateAsync` method. But maybe I'm missing something... Since this is one line of code, I believe that we can agree that we can add the ConfigureAwait(false) at least. The same for the previous one.

Since you said that "(...) they all don't need to be async" we change the return type to `ValueTask` them when we don't need to wait we don't pay the price for it.

Reply:
@pictos can you please show me what you mean. How can we avoid "async" in this particular case?

Reply:
I don't think we can put configureAwait(false) here, because we want to return back to UI context as soon as validation is ended (for updating styles)

Reply:
I'll check out it here and play with it to have a better vision of the code

Reply:
ValueTask sounds good

Reply:
@AndreiMisiukevich I see now about your snippet, I missed the second line on my review here on Github, in the IDE I could see it. Shame on me.

Yeah, we can't avoid `async` here. I tried with `ConfigureAwait(false)` and works. If you think that is valuable...

```csharp
protected override async Task<bool> ValidateAsync(object value, CancellationToken token)
			=> (await base.ValidateAsync(value, token).ConfigureAwait(false))
				&& Uri.IsWellFormedUriString(value?.ToString(), UriKind);
```

Reply:
updated)